sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java

```diff
@@ -102,14 +103,14 @@
   // The timezone conversion to apply to int96 timestamps. Null if no conversion.
   private final ZoneId convertTz;
   private static final ZoneId UTC = ZoneOffset.UTC;
-  private final boolean rebaseDateTime;
+  private final String datetimeRebaseMode;
```

Comment:
it's very hard to use Scala enum in Java, so I use string instead.

Reply:
I don't think it affects perf, and the code will be less readable if we see `mode == 0` instead of `"LEGACY".equals(mode)`